platform/macos/app/MapDocument.m

```diff
@@ -642,12 +649,17 @@ - (BOOL)validateMenuItem:(NSMenuItem *)menuItem {
         return YES;
     }
     if (menuItem.action == @selector(showColorBuffer:)) {
-        BOOL enabled = self.mapView.debugMask & MGLMapDebugStencilBufferMask;
+        BOOL enabled = self.mapView.debugMask & (MGLMapDebugStencilBufferMask | MGLMapDebugDepthBufferMask);
         menuItem.state = enabled ? NSOffState : NSOnState;
         return YES;
     }
     if (menuItem.action == @selector(showStencilBuffer:)) {
-        BOOL enabled = self.mapView.debugMask & MGLMapDebugStencilBufferMask;
+        BOOL enabled = self.mapView.debugMask & (MGLMapDebugStencilBufferMask & ~MGLMapDebugDepthBufferMask);
```

Comment:
Are the stencil and depth buffers mutually exclusive, as implied by this logic? If so, setting the depth buffer flag should unset the stencil buffer flag and vice versa.